### PR TITLE
feat(CLI): Prune cache

### LINKED ISSUE #128: feat(CLI): Prune cache (both buildkit and midi cache)
## Description

We should provide a way to clear the cache.

### PR DESCRIPTION
**Pull request by gaocegege**
Fix #128 

Signed-off-by: Ce Gao <cegao@tensorchord.ai>

### CODE REVIEW

File: pkg/app/prune.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			Name:  "keep-storage",
+			Usage: "Keep data below this limit (in MB)",
+		},
+		&cli.StringSliceFlag{
+			Name:  "filter, f",
+			Usage: "Filter records",
+		},
+		&cli.BoolFlag{
+			Name:  "all",
+			Usage: "Include internal caches",
```
Comment (VoVAllen) [inline pkg/app/prune.go:42]: What's internal caches?

File: pkg/app/prune.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+			Name:  "keep-duration",
+			Usage: "Keep data newer than this limit",
+		},
+		&cli.Float64Flag{
+			Name:  "keep-storage",
+			Usage: "Keep data below this limit (in MB)",
+		},
+		&cli.StringSliceFlag{
+			Name:  "filter, f",
+			Usage: "Filter records",
```
Comment (VoVAllen) [inline pkg/app/prune.go:38]: What can be filtered?

File: pkg/app/prune.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	Name:  "prune",
+	Usage: "clean up build cache",
+	Flags: []cli.Flag{
+		&cli.DurationFlag{
+			Name:  "keep-duration",
+			Usage: "Keep data newer than this limit",
+		},
+		&cli.Float64Flag{
+			Name:  "keep-storage",
+			Usage: "Keep data below this limit (in MB)",
```
Comment (VoVAllen) [inline pkg/app/prune.go:34]: Does it work like an LRU cache?

Comment (gaocegege) [reply pkg/app/prune.go:34]: No, we do not guarantee it.

File: pkg/app/prune.go (hunk 2 above)
Comment (gaocegege) [reply pkg/app/prune.go:38]: It is the CLI flag in buildkit. You can filter records by fields like description, last used time. We should document it in the doc

File: pkg/app/prune.go (hunk 1 above)
Comment (gaocegege) [reply pkg/app/prune.go:42]: buildkit's internal cache and envd host cache like oh-my-zsh

**Comment by gaocegege** [comment]
PTAL again.

**Review by VoVAllen — APPROVED** [review]
LGTM. Please also update the docs once merged.

**Pull request merged**